File: clients/kpi-client/src/constants/LoaiTaiLieuRutTienKyQuyConstant.ts
import { createConstant } from "./Constant";

// Mã loại tài liệu đính kèm cho hồ sơ Rút tiền ký quỹ (RutTienKyQuy).
// Tham chiếu spec: Documents/Hệ thống online_2.docx — section 5.
// Spec liệt kê 2 loại:
//   - Đơn đề nghị
//   - Văn bản, tài liệu kèm theo
const LoaiTaiLieuRutTienKyQuyConstant = createConstant(
  {
    DonDeNghi:     "DonDeNghi",
    VanBanKemTheo: "VanBanKemTheo",
  } as const,
  {
    DonDeNghi:     { displayName: "Đơn đề nghị" },
    VanBanKemTheo: { displayName: "Văn bản, tài liệu kèm theo" },
  }
);

export default LoaiTaiLieuRutTienKyQuyConstant;

File: services/asset-service/Hinet.Service/Constant/LoaiTaiLieuRutTienKyQuyConstant.cs
using System.ComponentModel;

namespace Hinet.Service.Constant
{
    // Mã loại tài liệu đính kèm cho Rút tiền ký quỹ.
    // Mirror FE: Client/src/constants/LoaiTaiLieuRutTienKyQuyConstant.ts
    public class LoaiTaiLieuRutTienKyQuyConstant
    {
        [DisplayName("Đơn đề nghị")]
        public static string DonDeNghi => "DonDeNghi";

        [DisplayName("Văn bản, tài liệu kèm theo")]
        public static string VanBanKemTheo => "VanBanKemTheo";

        public static readonly HashSet<string> ALL = new()
        {
            DonDeNghi,
            VanBanKemTheo,
        };

        public static bool IsValid(string? code) =>
            !string.IsNullOrWhiteSpace(code) && ALL.Contains(code);
    }
}
